src/Microsoft.AspNetCore.Mvc.Core/RequireHttpsAttribute.cs

```diff
@@ -19,12 +19,29 @@ public class RequireHttpsAttribute : Attribute, IAuthorizationFilter, IOrderedFi
         /// Specifies whether a permanent redirect, <c>301 Moved Permanently</c>,
         /// should be used instead of a temporary redirect, <c>302 Found</c>.
         /// </summary>
-        public bool Permanent { get; set; }
+        public bool? Permanent { get; set; }
 
         /// <inheritdoc />
         public int Order { get; set; }
 
         /// <summary>
+        /// Initializes an instance of <see cref="RequireHttpsAttribute"/>.
+        /// </summary>
+        public RequireHttpsAttribute()
+        {
+            this.Permanent = null;
```

Comment:
This assignment is redundant. In fact, why even have these constructors? They were left out intentionally.